pom.xml

```diff
@@ -96,7 +96,7 @@
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-api</artifactId>
-      <version>1.4.1</version>
```

Comment:
This contradicts the API version bundled with Maven 3.1.1.

Reply:
Is that a problem?

Reply:
Logically yes. This is contract we guarantee to work from Maven 3.1.1. Resolver API 1.6.1. will arrive with Maven 4 only.

Reply:
When I compared 1.4.1 and 1.6.1 -api artifact is 100% binary and source compatible

Reply:
@slachiewicz That's correct. I still prefer consistency. Just my two cents.

Reply:
perhaps we should mark such dependencies as provided: at least it would explain why we half care (it's not the version we'll have at runtime)

for Maven 4: this is perhaps some case where we could remove one more version tag = "we want to make explicit that we use that library, but with the version that comes from transitive"

I also prefer consistency (with a comment explaining it, because it's not obvious from an external reader point of view)

Reply:
Let's see if I understand this. I'm thinking that:

1. When Maven runs to build a user project, it has already loaded maven-resolver-api before it loads this plugin, so it uses the version of maven-resolver-api that is already in the classpath. That could be 1.4.1 for Maven 3.1.1 or some other version with a different version of Maven. Either way, the version declared here is moot. Hence, provided.

2. When this plugin is compiled, though, javac places the version of maven-resolver-api specified here on the classpath. 

does that make sense?

Reply:
yes, that's it
the little precision is that it's not a question of classpath order, but a Maven core feature that just skips Maven core provided dependencies: it skips "exported artifacts" from https://maven.apache.org/ref/3.6.3/maven-core/core-extensions.html